cmd/common.go

```diff
@@ -19,15 +19,29 @@ func init() {
 var (
 	goBin      string
 	flutterBin string
+	dockerBin  string
 )
 
 func initBinaries() {
 	var err error
+	goAvailable := false
+	dockerAvailable := false
 	goBin, err = exec.LookPath("go")
-	if err != nil {
-		fmt.Println("hover: Failed to lookup `go` executable. Please install Go.\nhttps://golang.org/doc/install")
+	if err == nil {
+		goAvailable = true
+	}
+	dockerBin, err = exec.LookPath("docker")
+	if err == nil {
+		dockerAvailable = true
+	}
+	if !dockerAvailable && !goAvailable {
+		fmt.Println("hover: Failed to lookup `go` and `docker` executable. Please install one of them:\nGo: https://golang.org/doc/install\nDocker: https://docs.docker.com/install")
 		os.Exit(1)
 	}
+	if dockerAvailable && !goAvailable {
+		fmt.Println("hover: Using Docker instead of go for compiling, because looking up `go` executable failed.")
```

Comment:
I think this may be unexpected behavior for some users. At the same time it is nice if users can get started quickly. I would feel more comfortable if this "exception" of the normal/happy flow would have a question attached; "do you want to continue using docker cross-compile support?".
The initial println could also explain that to persist this behavior and avoid the question, the user should explicitly add `--docker`.
Alternatively, instead of asking a question and entering a build flow that wasn't initially specified in the `hover` commandline arguments, we could just tell the user to use `--docker` and exit. Adding `--docker` is trivial for the user, and after that it's just "arrow up, enter" anyway.